webapp/src/utils/user_agent.jsx

```diff
@@ -1,8 +1,35 @@
 // Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
 // See LICENSE.txt for license information.
 
+import {isMinimumServerVersion} from 'mattermost-redux/utils/helpers';
+
 const userAgent = window.navigator.userAgent;
 
 export function isDesktopApp() {
     return userAgent.indexOf('Mattermost') !== -1 && userAgent.indexOf('Electron') !== -1;
 }
+
+// Desktop app user agent examples:
+// Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Mattermost/4.3.0-develop Chrome/73.0.3683.121 Electron/5.0.10 Safari/537.36'
+// Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Mattermost/4.2.3 Chrome/61.0.3163.100 Electron/2.0.12 Safari/537.36
+export function getDesktopAppVersion() {
+    if (!isDesktopApp()) {
+        return null;
+    }
+
+    const beginIndex = userAgent.indexOf('Mattermost') + 'Mattermost/'.length;
+    const developDashIndex = userAgent.substring(beginIndex).indexOf('-');
+    const spaceIndex = userAgent.substring(beginIndex).indexOf(' ');
+    const first = Math.min(developDashIndex, spaceIndex);
+
```

Comment:
1. I know its annoying, but I think we need tests for this -- we're parsing with string indexes here, which is a magnet for errors.
2. Are we sure this works for all the possible desktop app versions?

Reply:
@cpoile Thanks for taking a look at this :+1:

1. I totally agree, not annoying at all

2. I pulled @deanwhillier into this review as he can shed light on the nuances of the desktop app. The code here certainly seems rickety at the moment.

Reply:
I took a quick look at the Desktop source and it doesn't look like we're setting the agent explicitly, which means Electron is doing so on our behalf. I think it's safe to say that it is consistent. That said, relying on the user agent might not be the best approach? 🤷‍♂️

For v4.3.0, I implemented a rudimentary communication bridge between the Desktop App and the Webapp to handle user activity updates from the Desktop app as well as for the Webapp to send desktop notification requests to the Desktop App for handling (because, many reasons). In doing this I added a mechanism for the Desktop app to register itself with the Webapp once a user has logged in, including passing its version number. That version is currently stored as a string in `window.desktop.version` if it is a valid semantic version (not the best spot long term). I added and am using the `semver` package to parse, validate and compare version numbers: https://www.npmjs.com/package/semver.

So, as of 5.16 (backported to lastest 5.9, 5.14 & 5.15), if the Desktop App v4.3.0+ loads the Webapp, the Webapp will have access to the Desktop Version code it is running in via `window.desktop.version` once the user has logged in and you can use `semver` to do comparisons as needed.

Reply:
@deanwhillier Your suggestion seems like a better approach because it is more within our control. I'm wondering if the error we show here will suffice (considering the webapp version also matters here). I think making the error as user-friendly as possible is the way to combat this.

Reply:
@mickmister, I guess that depends on which server/webapp version(s) the Jira plugin can run on? If it can run on servers that don't include the `window.desktop.version` property, then the ephemeral message would probably need to mention a server version in addition to the desktop app version needed for connecting in-app. Challenge is wording the message so as not to be too technical or verbose.

Maybe something like "The current version of your desktop app doesn't support connecting to Jira directly, please connect through your browser or [see here](tbd) for details on upgrading." The link could point to our docs with a more detailed description of what is needed to connect to Jira with the desktop app including version numbers of both the app and server. It's also easier to managed details like these in a doc vs in the plugin itself. Just a thought! Might want to run this by the PM?

Reply:
Excellent thinking! We can offload these complex explanations to the docs. @aaronrothschild What are your thoughts on this?

Reply:
@aaronrothschild Ping for this, I want to get your input on this before I move forward with next steps